tests/test_model_generator.py

```diff
@@ -3722,13 +3772,277 @@ def test_modelgen_save_reload_links_03(self):
+        self.assertEqual(tsl2.comp_opt_friend.id, tsl.opt_friend.id)
+        self.assertEqual(
+            tsl2.comp_opt_wprop_friend.id, tsl.opt_wprop_friend.id
+        )
+
+        # Change all links to Billie
+        tsl2.req_wprop_friend = default.TestSingleLinks.req_wprop_friend.link(
+            billie
+        )
+        tsl2.req_friend = billie
+        tsl2.opt_friend = billie
+        tsl2.opt_wprop_friend = default.TestSingleLinks.opt_wprop_friend.link(
+            billie
+        )
+
+        # Save the changes
+        self.client.sync(tsl2)
+
+        # After save, the main links should be updated to Billie
+        self.assertEqual(tsl2.req_wprop_friend.name, "Billie")
+        self.assertEqual(tsl2.req_friend.name, "Billie")
+        self.assertEqual(tsl2.opt_friend.name, "Billie")
+        self.assertEqual(tsl2.opt_wprop_friend.name, "Billie")
+
+        # But computed links should no longer be set after the save
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_req_wprop_friend' is not set"
+        ):
+            assert tsl2.comp_req_wprop_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_req_friend' is not set"
+        ):
+            assert tsl2.comp_req_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_opt_friend' is not set"
+        ):
+            assert tsl2.comp_opt_friend is not None  # access field
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_opt_wprop_friend' is not set"
+        ):
+            assert tsl2.comp_opt_wprop_friend is not None  # access field
+
+    def test_modelgen_save_reload_links_05(self):
+        from models import default
+
+        # Test that backlinks become unset after save operations
+        # User.groups is a computed backlink to UserGroup.users
+
+        # Fetch Alice with her groups backlink
+        alice = self.client.get(
+            default.User.select(
+                name=True,
+                groups=True,
+            ).filter(name="Alice")
+        )
+
+        # Alice should have her groups backlink available after fetch
+        # (even if empty, it should not be None)
+        self.assertIsNotNone(alice.groups)
+
+        # Create a new UserGroup with Alice as a member
+        gr = default.UserGroup(
+            name="Dancing Banana Enthusiasts", users=[alice]
+        )
+        self.client.sync(gr)
+
+        # After saving the group, Alice's groups backlink should be unset
+        # because Alice wasn't explicitly fetched with groups after the change
+        with self.assertRaisesRegex(AttributeError, "'groups' is not set"):
+            assert alice.groups is not None  # access field
+
+        with self.assertRaisesRegex(AttributeError, "'groups' is not set"):
+            a = [u for u in gr.users if u.id == alice.id][0]
+            assert a.groups is not None  # access field
```

Comment:
Similar to the previous case, `alice` has been explicitly fetched with the backlink groups. The Python object `alice` has been literally added to the new group's multi link `users`. So it is reachable by the sync from the root `gr`. Therefore the expectation is that the backlink of `alice` must be re-fetched because it was not "unset".

Reply:
ok I fixed this one